packages/loot-core/src/server/accounts/rules.ts

```diff
@@ -663,135 +664,96 @@ function execNonSplitActions(actions: Action[], transaction) {
-  const parentActions = actions.filter(action => !action.options?.splitIndex);
-  const childActions = actions.filter(action => action.options?.splitIndex);
-  const totalSplitCount =
-    actions.reduce(
-      (prev, cur) => Math.max(prev, cur.options?.splitIndex ?? 0),
-      0,
-    ) + 1;
-
-  let update = execNonSplitActions(parentActions, transaction);
-  if (totalSplitCount === 1) {
-    // No splits, no need to do anything else.
-    return update;
-  }
-
-  if (update.is_child) {
-    // Rules with splits can't be applied to child transactions.
-    return update;
-  }
+function getSplitRemainder(transactions) {
+  const { error } = recalculateSplit(groupTransaction(transactions));
+  return error ? error.difference : 0;
+}
 
-  const splitAmountActions = childActions.filter(
+function execSplitActions(actions: Action[], transaction) {
+  const splitAmountActions = actions.filter(
     action => action.op === 'set-split-amount',
   );
-  const fixedSplitAmountActions = splitAmountActions.filter(
-    action => action.options.method === 'fixed-amount',
-  );
-  const fixedAmountsBySplit: Record<number, number> = {};
-  fixedSplitAmountActions.forEach(action => {
-    const splitIndex = action.options.splitIndex ?? 0;
-    fixedAmountsBySplit[splitIndex] = action.value;
-  });
-  const fixedAmountSplitCount = Object.keys(fixedAmountsBySplit).length;
-  const totalFixedAmount = Object.values(fixedAmountsBySplit).reduce<number>(
-    (prev, cur: number) => prev + cur,
-    0,
-  );
-  if (
-    fixedAmountSplitCount === totalSplitCount &&
-    totalFixedAmount !== (transaction.amount ?? totalFixedAmount)
-  ) {
-    // Not all value would be distributed to a split.
-    return transaction;
-  }
 
-  const { data, newTransaction } = splitTransaction(
-    ungroupTransaction(update),
+  // Convert the transaction to a split transaction.
+  const { data } = splitTransaction(
+    ungroupTransaction(transaction),
     transaction.id,
   );
-  update = recalculateSplit(newTransaction);
-  data[0] = update;
   let newTransactions = data;
 
-  for (const action of childActions) {
-    const splitIndex = action.options?.splitIndex ?? 0;
-    if (splitIndex >= update.subtransactions.length) {
-      const { data, newTransaction } = addSplitTransaction(
-        newTransactions,
-        transaction.id,
-      );
-      update = recalculateSplit(newTransaction);
-      data[0] = update;
+  // Add empty splits, and apply non-set-amount actions.
+  // This also populates any fixed-amount splits.
+  actions.forEach(action => {
+    const splitTransactionIndex = (action.options?.splitIndex ?? 0) + 1;
+    if (splitTransactionIndex >= newTransactions.length) {
+      const { data } = addSplitTransaction(newTransactions, transaction.id);
       newTransactions = data;
     }
-    action.exec(update.subtransactions[splitIndex]);
-  }
+    action.exec(newTransactions[splitTransactionIndex]);
+  });
```

Comment:
The zeroth index of the splits is reserved for actions that apply to all splits. It will only contain regular actions so we skip it elsewhere in the splits functions.